src/main/java/io/nats/client/impl/SocketDataPortWithWriteTimeout.java

```diff
@@ -42,10 +42,12 @@ public void run() {
                     out.close();
                 }
                 catch (IOException ignore) {}
-                try {
-                    connection.forceReconnect();
-                }
-                catch (InterruptedException | IOException ignore) {}
+                connection.getExecutor().submit(() -> {
+                    try {
+                        connection.forceReconnect();
+                    }
+                    catch (IOException | InterruptedException ignore) {}
+                });
```

Comment:
I decided to run this on (yet another thread) since this whole Data Port implementation object will be closed during the forceReconnect